src/main/java/jenkins/branch/MultiBranchProjectFactory.java

```diff
@@ -59,7 +60,8 @@
      * @return a new uninitialized multibranch project (do not configure its {@link MultiBranchProject#getSourcesList} or call {@link MultiBranchProject#onCreatedFromScratch}), or null if unrecognized
      */
     @CheckForNull
-    public abstract MultiBranchProject<?,?> createProject(@Nonnull ItemGroup<?> parent, @Nonnull String name, @Nonnull List<? extends SCMSource> scmSources, @Nonnull Map<String,Object> attributes, @Nonnull TaskListener listener) throws IOException, InterruptedException;
+    public abstract MultiBranchProject<?,?> createProject(@Nonnull ItemGroup<?> parent, @Nonnull String name, @Nonnull List<? extends SCMSource> scmSources, @Nonnull OrphanedItemStrategy strategy,
+            @Nonnull Map<String,Object> attributes, @Nonnull TaskListener listener) throws IOException, InterruptedException;
```

Comment:
:bug: this is not binary-compatible.

Again it would be simpler to have a setter for the strategy. Then no change to `MultiBranchProjectFactory` implementations would be required at all, unless we specifically wanted to customize the behavior on a per-factory basis (which I do not foresee a need for).